python/tabby-eval/tabby_data_pipeline/assets.py

```diff
@@ -0,0 +1,87 @@
+import json
+import os, subprocess
+import modal
+
+import requests
+import pandas as pd
+
+import base64
+from io import BytesIO
+
+import matplotlib.pyplot as plt
+
+from typing import Dict, List
+
+from dagster import (
+    AssetExecutionContext,
+    MetadataValue,
+    asset,
+    get_dagster_logger,
+    op,
+    StaticPartitionsDefinition,
+    MultiPartitionsDefinition,
+    AssetIn,
+    Field,
+    Int,
+    file_relative_path
+)
+from . import analyze, create_csv
```

Comment:
there's no need to extract utility functions. You could just define the asset as python function and orgnize them in individual python files

Reply:
They are individual python files, but I have to import them and call them in the assets

Reply:
they don't have to be. They can be defined as `asset` directly.